docs/manual/nonempty-checker.tex

```diff
@@ -0,0 +1,57 @@
+\htmlhr
+\chapterAndLabel{NonEmpty Checker}{nonempty-checker}
+
+The NonEmpty Checker ensures that a Collection, Map, or an Iterator annotated as
+\code{@NonEmpty} contains at least one element.
+
+To run the NonEmpty Checker, supply the
+\code{-processor org.checkerframework.checker.nonempty.NonEmptyChecker}
+command-line option to javac.
+
+\sectionAndLabel{Type Qualifiers and Hierarchy}{nonempty-hierarchy}
+The NonEmpty Checker supports the following type qualifiers:
+\begin{description}
+
+    \item[\refqualclass{checker/nonempty/qual}{UnknownNonEmpty}]
+    Indicates that the value assigned to the annotated variable is not known to be a non-empty
+    collection, map, or iterator. It is the default type
+    qualifier, so it should rarely be written by a programmer.
+
+    \item[\refqualclass{checker/nonempty/qual}{NonEmpty}]
+    Indicates that a collection, map, or iterator value is non-empty.
+
+    \item[\refqualclass{checker/nonempty/qual}{PolyNonEmpty}]
+    indicates qualifier polymorphism.
+    For a description of qualifier polymorphism, see
+    Section~\ref{method-qualifier-polymorphism}.
+
+\end{description}
+
+\begin{figure}
+    \includeimage{nonempty}{3cm}
+    \caption{Type hierarchy for the NonEmpty type system.}
+    \label{fig-nonempty-hierarchy}
+\end{figure}
+
+Figure~\ref{fig-nonempty-hierarchy} shows the qualifier hierarchy in the NonEmpty type system.
+
+\subsectionAndLabel{Type-refinement rules in the NonEmpty Checker}{nonempty-type-refinement}
```

Comment:
These are implementation details.  I don't think a reader needs to know them.
It might also be confusing to a reader that other tests are missing.  Two examples are:
 * that for `containsKey`
 * that for the else branch after a test of `Collection.isEmpty()`

I suggest removing these.